### PR TITLE
Add docstring to print_tree

### PR DESCRIPTION
**Pull request by KronosTheLate**

### CODE REVIEW

File: src/DecisionTree.jl [hunk 1]
```diff
@@ -83,6 +83,31 @@ function print_tree(leaf::Leaf, depth=-1, indent=0; feature_names=nothing)
     println("$(leaf.majority) : $(ratio)")
 end
 
+"""
+       print_tree(tree::Node, depth=-1, indent=0; feature_names=nothing)
+
+Print a textual visualization of the given decision tree `tree`.
+In the example output below, the top node look at if
```
Comment (ablaom) [inline src/DecisionTree.jl:90]: Minor grammatical suggestion:

```suggestion
In the example output below, the top node considers whether 
```

**Review by ablaom — CHANGES_REQUESTED** [review]
Otherwise looks good to me. Thanks @KronosTheLate for this addition.

**Comment by KronosTheLate** [comment]
> Otherwise looks good to me. Thanks @KronosTheLate for this addition.

Great. I was not certain if my interpretation of the tree was correct, which is the reason I found the need to try explaining it. And just hoping that I would be corrected if I did it wrong.

File: src/DecisionTree.jl [hunk 2]
```diff
@@ -83,6 +83,31 @@ function print_tree(leaf::Leaf, depth=-1, indent=0; feature_names=nothing)
     println("$(leaf.majority) : $(ratio)")
 end
 
+"""
+       print_tree(tree::Node, depth=-1, indent=0; feature_names=nothing)
+
+Print a textual visualization of the given decision tree `tree`.
+In the example output below, the top node considers whether 
+"Feature 3" is above or below the threshold -28.156052806422238.
+If the value of "Feature 3" is below the threshold for some input to be classified,
```
Comment (ablaom) [inline src/DecisionTree.jl:92]: To be even clearer (relevant for discrete features):

```suggestion
If the value of "Feature 3" is strictly below the threshold for some input to be classified, 
```

File: src/DecisionTree.jl [hunk 3]
```diff
@@ -83,6 +83,31 @@ function print_tree(leaf::Leaf, depth=-1, indent=0; feature_names=nothing)
...
+"""
+       print_tree(tree::Node, depth=-1, indent=0; feature_names=nothing)
+
+Print a textual visualization of the given decision tree `tree`.
+In the example output below, the top node considers whether 
+"Feature 3" is above or below the threshold -28.156052806422238.
+If the value of "Feature 3" is below the threshold for some input to be classified, 
+we move to the `L->` part underneath, which is a node 
+looking at if "Feature 2" is above or below -161.04351901384842.
+If the value of "Feature 2" is below the threshold for some input to be classified,
```
Comment (ablaom) [inline src/DecisionTree.jl:95]:
```suggestion
If the value of "Feature 2" is strictly below the threshold for some input to be classified, 
```

**Comment by ablaom** [comment]
> I was not certain if my interpretation of the tree was correct, 

Just to be sure, I checked in another example. All good.

**Review by ablaom — CHANGES_REQUESTED** [review]
All good after marked changes.

**Review by ablaom — APPROVED** [review]
Good to go.

**Pull request merged**